### PR TITLE
Lazyload navigation screens

### PR DESCRIPTION
**Pull request by koxen**
reduce startup overhead

#### Summary
Lazy load navigation screens to make startup faster.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Added or updated unit tests (required for all new features)
- [ ] All new/modified APIs include changes to [mattermost-redux](https://github.com/mattermost/mattermost-redux) (please link)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates

#### Device Information
This PR was tested on: [Device name(s), OS version(s)] 
Samsung S7, Android 7.0

### CODE REVIEW

**Comment by koxen** [comment]
While the performance improvement isn't as big as originally expected, I would still like to merge this as an improvement. Thoughts?

**Comment by koxen** [comment]
@enahum this one is also ready for review; this is going to be helpful when we switch to `react-native unbundle --indexed-unbundle`; AFAIK they fixed it very recently on the Android side - in 0.54

**Comment by enahum** [comment]
I don't want to rush with this one, release for 1.7 is coming up and definitely don't want to have this PR in this late in the game, perhaps we can try this after we cut final, are you ok with that?

**Comment by koxen** [comment]
sure thing, thanks for details :)

**Comment by miguelespinoza** [comment]
@enahum @koxen it would be great if this could get tested and merged. 
I have a working prototype that reduces cold start to ~ 2 seconds and this played a role

**Comment by enahum** [comment]
@miguelespinoza does your prototype uses the stack RN or a fork?

can you share more? I haven't been able to make this PR work

**Comment by miguelespinoza** [comment]
@enahum it's a rewrite of mattermost.js
I should have something early next week.  A couple of minor things here and there to fix.

> I haven't been able to make this PR work
@enahum what issues are you seeing?  I didn't do anything special to get this working.  Any errors, or unable to run app?

File: app/screens/index.js [hunk 1]
```diff
@@ -62,43 +21,43 @@ function wrapWithContextProvider(Comp, excludeEvents = true) {
...
-    Navigation.registerComponent('Search', () => wrapWithContextProvider(Search), store, Provider);
-    Navigation.registerComponent('SelectServer', () => wrapWithContextProvider(SelectServer), store, Provider);
-    Navigation.registerComponent('SelectTeam', () => wrapWithContextProvider(SelectTeam), store, Provider);
-    Navigation.registerComponent('Settings', () => wrapWithContextProvider(Settings), store, Provider);
-    Navigation.registerComponent('SSO', () => wrapWithContextProvider(SSO), store, Provider);
-    Navigation.registerComponent('Table', () => wrapWithContextProvider(Table), store, Provider);
-    Navigation.registerComponent('TableImage', () => wrapWithContextProvider(TableImage), store, Provider);
-    Navigation.registerComponent('Thread', () => wrapWithContextProvider(Thread), store, Provider);
-    Navigation.registerComponent('UserProfile', () => wrapWithContextProvider(UserProfile), store, Provider);
+    Navigation.registerComponent('About', () => wrapWithContextProvider(require('app/screens/about').default), store, Provider); //eslint-disable-line global-require
```
Comment (enahum) [inline app/screens/index.js]: You could do this here
```js
export function registerScreens(store, Provider) {
    /* eslint-disable global-require */
   ...
}
```

that way you do not need to add the `eslint-disable-line` in each line

**Comment by saturninoabril** [comment]
@koxen Could you share your metrics just to see how this improves like baseline vs new startup time (both for iOS and Android)?
Also, I would appreciate to know some context on how import vs require improves lazy loading.

**Comment by enahum** [comment]
close in favor #1598

**Pull request closed without merging**